File: backend/src/main/java/com/gabrielluciano/blog/dto/post/PostResponse.java
package com.gabrielluciano.blog.dto.post;

import com.gabrielluciano.blog.dto.tag.TagResponse;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PostResponse {

    private Long id;
    private String title;
    private String subtitle;
    private String content;
    private String metaTitle;
    private String metaDescription;
    private String slug;
    private String imageUrl;
    private Boolean published;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private LocalDateTime publishedAt;
    private Set<TagResponse> tags;
}

File: frontend/src/app/models/post/postResponse.ts
import { TagResponse } from '../tag/tagResponse';
import { UserResponse } from '../user/userResponse';

export interface PostReponse {
  id: number;
  title: string;
  subtitle: string;
  content: string;
  metaTitle: string;
  metaDescription: string;
  slug: string;
  imageUrl: string;
  published: boolean;
  createdAt: string;
  updatedAt: string;
  publishedAt: string;
  tags: TagResponse[];
  author: UserResponse;
}
